src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -0,0 +1,391 @@
+    INTEGER("integer", 0)
+    ;
+
+    public enum Is
+    {
+        REQUIRED,
+        OPTIONAL
+    }
+
+    @Nonnull private final String usageValue;
+    private final int allowedSpaceCount;
+
+    ActionArgumentType(@Nonnull final String usageValue, final int allowedSpaceCount)
+    {
+        this.usageValue = usageValue;
+        this.allowedSpaceCount = allowedSpaceCount;
+    }
+
+    public String getUsageValue()
+    {
+        return usageValue;
+    }
+
+    public int allowedSpaceCount()
+    {
+        return allowedSpaceCount;
+    }
+
+    private List<String> getOnlinePlayerNames(@NotNull final MinecraftServer server)
+    {
+        final String[] onlinePlayerNames = server.getOnlinePlayerNames();
+        return Arrays.asList(onlinePlayerNames);
+    }
+
+    private List<String> getColonyIdStrings()
+    {
+        final List<Colony> colonyList = ColonyManager.getColonies();
+        final List<String> colonyIdList = new ArrayList<>(colonyList.size());
+        for (final Colony colony : colonyList)
+        {
+            colonyIdList.add(String.valueOf(colony.getID()));
+        }
+        return colonyIdList;
+    }
+
+    private List<String> getCitizenNames()
+    {
+        final List<Colony> colonyList = ColonyManager.getColonies();
+        final List<String> citizenNameList = new ArrayList<>();
+        for (final Colony colony : colonyList)
+        {
+            final List<CitizenData> citizenDataList = colony.getCitizenManager().getCitizens();
+            for (final CitizenData citizenData : citizenDataList)
+            {
+                citizenNameList.add(citizenData.getName());
+            }
+        }
+        return citizenNameList;
+    }
+
+    public List<String> getTabCompletions(@NotNull final MinecraftServer server,
+            @Nullable final BlockPos pos,
+            final String potentialArgumentValue)
+    {
+        switch (this)
+        {
+            case INTEGER:
+                return Collections.emptyList();
+            case BOOLEAN:
+                return Arrays.asList(new String[] {"true", "false"});
+            case COORDINATE_X:
+            case COORDINATE_Y:
+            case COORDINATE_Z:
+                return getCoordinateTabCompletions(pos, potentialArgumentValue);
+            case PLAYER:
+                final List<String> playerNameStrings = getOnlinePlayerNames(server);
+                return playerNameStrings.stream().filter(k -> k.startsWith(potentialArgumentValue)).collect(Collectors.toList());
+            case COLONY:
+                return getColonyTabCompletions(potentialArgumentValue);
+            case CITIZEN:
+                return getCitizenTabCompletions(potentialArgumentValue);
+            default:
+                throw new IllegalStateException("Unimplemented ActionArgumentType tab completion");
+        }
+    }
+
+    private List<String> getCoordinateTabCompletions(@Nullable final BlockPos pos, final String potentialArgumentValue)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)